workflow/controller/agent.go

```diff
@@ -257,6 +258,10 @@ func (woc *wfOperationCtx) createAgentPod(ctx context.Context) (*apiv1.Pod, erro
 				return existing, nil
 			}
 		}
+		if errorsutil.IsTransientErr(err) {
+			woc.requeue()
+			return created, nil
```

Comment:
I would think that `created` would be nil if we get here, right? Or am I wrong?

Reply:
Okay, never mind - I see you're just being complete, in the case that some transient error does result in a non-nil Pod for some reason.